### PR TITLE
Add Storybook integration in flow-library

### LINKED ISSUE #2262: Integrate `storybook` with yoshi-flow-library
# 🙋 Feature Request

When creating a library which has react components, it makes total sense to use storybook to work on your components with a development server, it's also good as a showcase of all of the possible states of the component.

It should be a feature that you get out of the box when using `yoshi-flow-library`.

There is an [integration for storybook](https://github.com/wix/yoshi/pull/1861) in yoshi's dev-environment, so we just need to connect it with the flow.

I believe that this would be a very cool feature that should be used by many library authors in Wix.

## Context

That feature was specifically requested by @borislit ❤️

### PR DESCRIPTION
**Pull request by daliusd**
Closes #2262

### CODE REVIEW

File: test/fixtures/flow-library/base-template/dev/index.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>generated test</title>
+        <meta charset="utf-8">
+    </head>
+    <body>
+        <script src="http://localhost:3300/generated.umd.js"></script>
+    </body>
+</html>
```
Comment (amitdahan) [inline test/fixtures/flow-library/base-template/dev/index.html:10]: Pretty sure this file can be deleted

File: test/fixtures/flow-library/base-template/dev/server.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const path = require('path');
+const express = require('express');
+
+const app = express();
+
+app.get('/', (_req, res) => {
+  res.sendFile(path.join(__dirname, './index.html'));
+});
+
+app.listen(process.env.PORT);
```
Comment (amitdahan) [inline test/fixtures/flow-library/base-template/dev/server.js:10]: Same, the `dev` dir has to be used manually (by manually serving the html file which points to the local CDN you DO get from `yoshi-library start`.
[Library flow doesn't serve html](https://github.com/wix/yoshi/issues/2263) yet.
Which I guess means it's definitely not used in tests that run the `yoshi-library` cli :)

File: test/typescript/features/storybook/flow-library-happy-flow/storybook-flowlibraryhappyflow.test.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      await page.goto(
+        'http://localhost:9009/iframe.html?selectedKind=Components&selectedStory=Basic',
+      );
+      await page.waitForSelector('#component');
+      const result = await page.$eval('#component', (elm) => elm.textContent);
+      expect(result).toBe('Component in Storybook');
+    },
+    { waitForStorybook: true },
+  );
+});
```
Comment (amitdahan) [inline test/typescript/features/storybook/flow-library-happy-flow/storybook-flowlibraryhappyflow.test.ts:20]: Wdyt of naming it `storybook-library-happyflow.test.ts` instead?
I think it's more consistent with the other storybook happyflow test.

**Comment by amitdahan** [comment]
Oh one last thing, nothing that prevents merging this.

Running storybook currently requires users to both enable (`storybook: true`) and install `yoshi-storybook-dependencies`.

When the config option is enabled the user sees an error to let him know he needs to install the dependency pack.

Maybe we could improve by either:
1. Removing the need for the config option entirely, and just provide storybook if the dependency pack is installed.
1. Add instructions to the docs to install it along with turning the config on.

Wdyt @daliusd & @ranyitz ?

**Comment by daliusd** [comment]
> Oh one last thing, nothing that prevents merging this.
> 
> Running storybook currently requires users to both enable (`storybook: true`) and install `yoshi-storybook-dependencies`.
> 
> When the config option is enabled the user sees an error to let him know he needs to install the dependency pack.
> 
> Maybe we could improve by either:
> 
> 1. Removing the need for the config option entirely, and just provide storybook if the dependency pack is installed.
> 2. Add instructions to the docs to install it along with turning the config on.
> 
> Wdyt @daliusd & @ranyitz ?

I have updated documentation as of now.

**Comment by amitdahan** [comment]
Hey @daliusd! I pushed [a PR](https://github.com/daliusd/yoshi/pull/1) to fix conflicts :D

**Pull request merged**